tests/test-webapps/test-servlet-spec/test-spec-webapp/pom.xml

```diff
@@ -127,6 +126,14 @@
                   <overWrite>true</overWrite>
                   <outputDirectory>${project.build.directory}/lib/jndi</outputDirectory>
                 </artifactItem>
+                <artifactItem>
```

Comment:
If the jetty-util-9.3.0.RC0 jar is not a direct dependency, then you won't be able to run the ClassLoader test via doing "mvn jetty:run".

Reply:
The direct dependency is the problem for everything else.
It cannot be a direct dependency in the reactor or we hit all of the normal day to day jetty-util issues all over again.
I would say this concern is a flaw in jetty-maven-plugin, in that it doesn't take into account generated `WEB-INF/lib` content, which it should.

Reply:
@janbartel how does the jetty-maven-plugin handle jars that are directly added to `WEB-INF/lib` and not listed as maven dependencies?

Reply:
The jetty-maven-plugin has only ever dealt with direct/transitive dependencies. For the millions of other ways you can stuff things into a war file we would use the "mvn jetty:run-war" goal instead.

Reply:
So is it acceptable that to run this particular test from the `jetty-maven-plugin` that `jetty:run-war` is used?

Reply:
You can use the `<excludedGoals>` configuration parameter for the jetty-maven-plugin to list goals that are known not to work.

Reply:
The fact that `mvn jetty:run` doesn't handle the standard `generated-*` and `prepare-*` phases from maven is a different problem than this issue / PR is trying to solve.
I'll open a new Issue for that.

Reply:
Regardless of #4422 can we list known unsupported goals as `<excludedGoals>` in this PR at the very least.

But ultimately, I think the solution to this problem is to stop using util as the jar for this example.    We just need to find some other random unrelated jar from maven that has two versions.  We make one a dependency of the webapp.    Then when we want to run the test, we put a different version on the server classpath, protect it as a server class and then test that the webapp can still load it's own version.

Reply:
oh ideally the 'random other jar' should be something from eclipse so QA is easy.